lib/bap_disasm/bap_disasm_reconstructor.ml

```diff
@@ -62,7 +66,10 @@ let reconstruct name roots cfg =
                       if Block.equal n entry
                       then Cfg.Node.insert n t
                       else return t)) in
-          Symtab.add_symbol syms (name,entry,cfg))
+          let fn = name,entry,cfg in
+          let symtab = Symtab.add_symbol syms fn in
+          Seq.fold (find_callers entry) ~init:symtab
+            ~f:(fun syms a -> Symtab.add_call syms a fn))
```

Comment:
This is a wrong assumption, that all predecessors of an entry block are callers. A function can have a back edge that points to the entry block. E.g.,



```
digraph {
    entry -> check;
    check -> exit;
    check -> entry;
}
```